### PR TITLE
[Lean] Rename user-facing options from Aeneas to Lean

### PR DESCRIPTION
**Pull request by zhassan-aws**
This is a follow-up on #3514. As @celinval suggested ([here](https://github.com/model-checking/kani/pull/3514#discussion_r1790949304)), renaming all user-facing options from "Aeneas" to Lean.

Inside the Kani compiler which is only concerned with producing LLBC (and doesn't know about Lean/Aeneas), I'm renaming the relevant feature from `aeneas` to `llbc`, and the backend is now called the LLBC backend as opposed to the Aeneas backend.

Towards #3585 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: kani-compiler/src/args.rs [hunk 1]
```diff
@@ -87,7 +87,7 @@ pub struct Arguments {
     /// Option name used to select which backend to use.
     #[clap(long = "backend", default_value_t = BackendOption::CProver)]
     pub backend: BackendOption,
-    /// Print the final LLBC file to stdout. This requires `-Zaeneas`.
+    /// Print the final LLBC file to stdout.
```
Comment (celinval) [inline kani-compiler/src/args.rs:90]: Why did you remove this comment?

**Review by celinval — APPROVED** [review]
Thanks!

File: kani-compiler/src/args.rs (hunk 1 above)
Comment (zhassan-aws) [reply kani-compiler/src/args.rs:90]: My understanding is that those requires comments should be in the kani-driver help, and not in the Kani compiler one and the kani-driver option already mentions the requirement:

https://github.com/model-checking/kani/pull/3630/files#diff-6f15b74109fec7be3b13bdc2f8c01481d82be82f413c39a7f54d53f1990c6454R280

None of the other Kani compiler options that require an unstable flag (e.g. `--check_coverage`, `--ub-check`) mention the unstable flag requirement in their help message.
